src/decays/HiggsTools_interface.cpp

```diff
@@ -0,0 +1,189 @@
+
+         auto effc = HP::NeutralEffectiveCouplings {};
+         // quarks
+         effc.dd = std::abs(sm_input[0].dd) > 0 ? el.dd/sm_input[0].dd.real() : 0.;
+         effc.uu = std::abs(sm_input[0].uu) > 0 ? el.uu/sm_input[0].uu.real() : 0.;
+         effc.ss = std::abs(sm_input[0].ss) > 0 ? el.ss/sm_input[0].ss.real() : 0.;
+         effc.cc = std::abs(sm_input[0].cc) > 0 ? el.cc/sm_input[0].cc.real() : 0.;
+         effc.bb = std::abs(sm_input[0].bb) > 0 ? el.bb/sm_input[0].bb.real() : 0.;
+         effc.tt = std::abs(sm_input[0].tt) > 0 ? el.tt/sm_input[0].tt.real() : 0.;
+         // leptons
+         effc.ee = std::abs(sm_input[0].ee) > 0 ? el.ee/sm_input[0].ee.real() : 0.;
+         effc.mumu = std::abs(sm_input[0].mumu) > 0 ? el.mumu/sm_input[0].mumu.real() : 0.;
+         effc.tautau = std::abs(sm_input[0].tautau) > 0 ? el.tautau/sm_input[0].tautau.real() : 0.;
+         // gauge bosons
+         effc.WW = std::abs(sm_input[0].WW) > 0 ? el.WW/sm_input[0].WW : 0.;
+         effc.ZZ = std::abs(sm_input[0].ZZ) > 0 ? el.ZZ/sm_input[0].ZZ : 0.;
+         effc.gamgam = std::abs(sm_input[0].gamgam) > 0 ? el.gamgam/sm_input[0].gamgam : 0.;
+         effc.Zgam = std::abs(sm_input[0].Zgam) > 0 ? el.Zgam/sm_input[0].Zgam : 0.;
+         effc.gg = std::abs(sm_input[0].gg) > 0 ? el.gg/sm_input[0].gg : 0.;
+
+         effectiveCouplingInput(s, effc, HP::ReferenceModel::SMHiggsEW, calcggH, calcHgamgam);
+         // effective coupligs are defined as sqrt(Gamma CP-even) + I sqrt(Gamma CP-odd)
+         // so taking a norm gives a total partial width
+         s.setDecayWidth(HP::Decay::emu,   std::norm(el.emu));
+         s.setDecayWidth(HP::Decay::etau,  std::norm(el.etau));
+         s.setDecayWidth(HP::Decay::mutau, std::norm(el.mutau));
+         // set total width to the one computed by FD as HiggsTools doesn't calculate
+         // some decays of Higgs at all, e.g. H -> Ah Z
+         if (el.width > s.totalWidth()) {
+            s.setDecayWidth("NP", "NP", el.width - s.totalWidth());
+         }
+      }
+   }
+
+   for (auto const& el : bsm_input2) {
+      auto& Hpm = pred.addParticle(HP::BsmParticle(el.particle, HP::ECharge::single));
+      // set mass
+      Hpm.setMass(el.mass);
+
+      // set H± decays
+      if (el.width > 0) {
+         Hpm.setTotalWidth(el.width);
+         Hpm.setBr(HP::Decay::enu, el.brenu);
+         Hpm.setBr(HP::Decay::munu, el.brmunu);
+         Hpm.setBr(HP::Decay::taunu, el.brtaunu);
+         Hpm.setBr(HP::Decay::tb, el.brtb);
+         Hpm.setBr(HP::Decay::cs, el.brcs);
+         Hpm.setBr(HP::Decay::WZ, el.brWZ);
+         Hpm.setBr(HP::Decay::Wgam, el.brWgam);
+      }
+
+      // production
+      // pp->ttbar, t(tbar)->Hp b
+      double ppHpmtb_xsec = HP::EffectiveCouplingCxns::ppHpmtb(HP::Collider::LHC13, el.mass, el.cHpmtbR, el.cHpmtbL, el.brtHpb);
+      Hpm.setCxn(HP::Collider::LHC13, HP::Production::Hpmtb, ppHpmtb_xsec);
+   }
+
+
+
+   // HiggsBounds
+   struct stat buffer;
```

Comment:
`sys/stat.h` is UNIX-specific. I think an OS-independent solution would be better, like so:
~~~.cpp
#include <boost/filesystem.hpp>

if (!boost::filesystem::exists("myfile.txt")) {
   ...
}
~~~

Reply:
Now that I think of this, since `HiggsTools` interface requires C++17, we can use a C++17 `filesystem` module.